internal/rpmtree.bzl

```diff
@@ -67,17 +67,29 @@ def _rpm2tar_impl(ctx):
 
     return [DefaultInfo(files = depset([ctx.outputs.out]))]
 
-def _expand_path(files):
-    return [x.path for x in files]
-
 def _tar2files_impl(ctx):
     args = ctx.actions.args()
 
     args.add_all(["tar2files", "--file-prefix", ctx.attr.prefix, "--input", ctx.files.tar[0]])
-    args.add_all([ctx.outputs.out], map_each = _expand_path)
+
+    file_paths_to_extract = [f.path for f in ctx.outputs.out]
+    # Avoid filelist for RPMs with few files to minimize write overhead.
+    if len(file_paths_to_extract) > 100:
```

Comment:
(1) can this be extracted as a constant
(2) can the threshold be more like 10 instead of 100?  From an ergonomic standpoint, having 100 files on the commandline is terrible as well